modules/local/samtools/reindex_bam/main.nf

```diff
@@ -0,0 +1,57 @@
+/**
+ * The aim of this process is to re-index the bam file without the duplicate, supplementary, unmapped etc, for goleft/indexcov
+ * It creates a BAM containing only a header (so indexcov can get the sample name) and a BAM index were low quality reads, supplementary etc, have been removed
+ */
+process SAMTOOLS_REINDEX_BAM {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.20--h50ea8bc_0' :
+        'biocontainers/samtools:1.20--h50ea8bc_0' }"
+
+    input:
+    tuple val(meta), path(input), path(input_index)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(fai)
+
+    output:
+    tuple val(meta), path("${meta.id}.reindex.bam"), path("${meta.id}.reindex.bam.bai"),emit: output
```

Comment:
Looks like indexcov is also supporting CRAM files. Is there a reason to first convert them to bam here and not use also CRAM?

Reply:
Oh sorry missed that.  merging this then now.